### PR TITLE
[1.11] Add a way to modify Entity AI

### PR DESCRIPTION
**Pull request by Madsthunder**
Added a way to modify Entity AI through an event. Works with any Entity as long as it has an instance of EntityAITasks.

### CODE REVIEW

File: patches/minecraft/net/minecraft/entity/EntityLiving.java.patch [hunk 1]
```diff
@@ -1,14 +1,22 @@
 --- ../src-base/minecraft/net/minecraft/entity/EntityLiving.java
 +++ ../src-work/minecraft/net/minecraft/entity/EntityLiving.java
-@@ -171,6 +171,7 @@
+@@ -108,6 +108,7 @@
+ 
+     protected void func_184651_r()
```
Comment (liach) [inline patches/minecraft/net/minecraft/entity/EntityLiving.java.patch:5]: What is this named?

**Comment by liach** [comment]
You need to fix the merge conflict.

File: patches/minecraft/net/minecraft/entity/EntityLiving.java.patch (hunk 1 above)
Comment (williewillus) [reply patches/minecraft/net/minecraft/entity/EntityLiving.java.patch:5]: you can use the bot yourself to find out :P

**Comment by williewillus** [comment]
why are you calling the superclass initialization after the subclass initialization?

**Comment by liach** [comment]
I would suggest modifying their AI only when they are added to the world. The patch is too big and meaningless.

**Comment by williewillus** [comment]
I don't see much of a point to this to be honest ,we can already listen to entity spawn events and edit the task list using reflection. the patch is too big for the value it returns

**Comment by liach** [comment]
Or to patch after the call of `func_184651_r()`

File: patches/minecraft/net/minecraft/entity/EntityLiving.java.patch (hunk 1 above)
Comment (simon816) [reply patches/minecraft/net/minecraft/entity/EntityLiving.java.patch:5]: I recommend using this extension if using Chrome
https://chrome.google.com/webstore/detail/mcp-srg-mapper/ijjbogjfmngomjdmnabifgckbonmcebh

**Comment by simon816** [comment]
@williewillus 
> why are you calling the superclass initialization after the subclass initialization?

Probably to allow vanilla to add it's AI tasks before the event is fired

**Comment by Madsthunder** [comment]
Ah, didn't notice those events

**Pull request closed without merging**